### PR TITLE
Sprk 270

### PR DESCRIPTION
**Pull request by thijssnelleman**
Added in validation to be done by Configurator instead of java SMAC internal. Took massive structural changes for SMAC and  completely isolated SMACv2 into a configurator subclass. Tested flake8, pytest and configuration.sh in examples.

### CODE REVIEW

File: CLI/run_configured_solver.py [hunk 1]
```diff
@@ -7,21 +7,20 @@
 
 from runrunner.base import Runner
 
-import global_variables as sgh
+import global_variables as gv
```
Comment (jeroenrook) [inline CLI/run_configured_solver.py:10]: As this is a relative import change to `.global_variables`. Additionally, this type of import does not pave the way to use the CLI  outside of the sparkle base directory. Please change this if possible.

File: Components/smac-v2.10.03-master-778/smac_target_algorithm.py [hunk 2]
```diff
@@ -31,14 +32,15 @@
     # 2. Build Run Solver call
     runsolver_binary = solver_dir / "runsolver"
     log_timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
-    runsolver_watch_data_path = Path(f"{str(Path.cwd())}/runsolver_{log_timestamp}.log")
+    runsolver_watch_data_path = Path(f"runsolver_{random.randint(0, 10000000000)}_{log_timestamp}.log")
+    runsolver_var_data_path = Path(f"runsolver_{random.randint(0, 10000000000)}_{log_timestamp}.var")
```
Comment (jeroenrook) [inline Components/smac-v2.10.03-master-778/smac_target_algorithm.py:36]: Neater: `int(random.getrandbits(32))`

File: Examples/Resources/Solvers/PbO-CCSAT-Generic/sparkle_solver_wrapper.py [hunk 3]
```diff
@@ -32,9 +32,20 @@
 
 # Construct call from args dictionary
 params = []
-for key in args:
-    if args[key] is not None:
-        params.extend(["-" + str(key), str(args[key])])
+if "config_path" in args:
```
Comment (jeroenrook) [inline Examples/Resources/Solvers/PbO-CCSAT-Generic/sparkle_solver_wrapper.py:35]: This code block looks quite complicated. Is the user required to copy this themselves into their own wrapper and is this a repetative thing? If so I suggest to make a wrapper helper in the sparkle package which users can call to obtain the configuration string, like
```
import sparkle.wrapper_help as wrapper

config = wrapper.get_config(args, as_list=True, keyprefix="-")
```

File: Settings/sparkle_settings.ini [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [general]
 objective = RUNTIME:PAR10
-configurator = smac_v2
+configurator = SMACv2
```
Comment (jeroenrook) [inline Settings/sparkle_settings.ini:3]: Can we make this SMAC2? This is how we pronounce it when talking with other people about it

File: sparkle/configurator/configuration_scenario.py [hunk 5]
```diff
@@ -82,22 +82,21 @@ def create_scenario(self: ConfigurationScenario, parent_directory: Path) -> None
     def _set_paths(self: ConfigurationScenario, parent_directory: Path) -> None:
```
Comment (jeroenrook) [inline sparkle/configurator/configuration_scenario.py:82]: Add description

File: sparkle/configurator/configurator.py [hunk 6]
```diff
@@ -4,152 +4,82 @@
...
-        (self.configurator_path / "tmp").mkdir(exist_ok=True)
 
-        self.objectives = sgh.settings.get_general_sparkle_objectives()
+        self.objectives = gv.settings.get_general_sparkle_objectives()
         if len(self.objectives) > 1 and not self.multiobjective:
             print("Warning: Multiple objectives specified but current configurator "
                   f"{self.configurator_path.name} only supports single objective. "
                   f"Defaulted to first specified objective: {self.objectives[0].name}")
 
     def configure(self: Configurator,
```
Comment (jeroenrook) [inline sparkle/configurator/configurator.py:53]: To all abstract methods (those who get functionality by the implementation), please add ` @abstractmethod` which you can get from: `from abc import abstractmethod`

File: CLI/run_configured_solver.py (hunk 1 above)
Comment (thijssnelleman) [reply CLI/run_configured_solver.py:10]: I do agree, but I think global_variables should be in the CLI dir at some point (Once we have decoupled the sparkle/ dir from global_variables). the .global_variables currently does not work (Linter/Python does not understand the pathing)

File: Examples/Resources/Solvers/PbO-CCSAT-Generic/sparkle_solver_wrapper.py (hunk 3 above)
Comment (thijssnelleman) [reply Examples/Resources/Solvers/PbO-CCSAT-Generic/sparkle_solver_wrapper.py:35]: Agreed, will write a ticket

File: Settings/sparkle_settings.ini (hunk 4 above)
Comment (thijssnelleman) [reply Settings/sparkle_settings.ini:3]: Done

File: CLI/run_configured_solver.py (hunk 1 above)
Comment (thijssnelleman) [reply CLI/run_configured_solver.py:10]: Writing ticket

**Pull request merged**